### ISSUE TITLE
Unable to find file or target named 'src/mmap_disk_job.cpp'

### ISSUE THREAD

**Issue by NickHo99** [userdocs/qbittorrent-nox-static#76]
OS: Debian 11
download and build commands:
```
git clone --depth 1 --branch release-4.4.0_v2.0.5 https://github.com/userdocs/qbittorrent-nox-static
./qbittorrent-nox-static.sh -qt release-4.4.0 -lt v2.0.5 -bv 1.77.0 all
```
The command worked fine two weeks ago, but today I got the error as below:
```
patching file include/libtorrent/aux_/allocating_handler.hpp
patch unexpectedly ends in middle of line
Hunk #1 succeeded at 124 with fuzz 1.
CXXFLAGS = -std=c++17 -static -w  -I/root/qbittorrent-nox-static/qbt-build/include
LDFLAGS = -static -L/root/qbittorrent-nox-static/qbt-build/lib
OS = LINUX
building boost from source directory:  /root/qbittorrent-nox-static/qbt-build/boost
Performing configuration checks

    - default address-model    : 64-bit [1]
    - default architecture     : x86 [1]
error: Unable to find file or target named
error:     'src/mmap_disk_job.cpp'
error: referred to from project at
error:     '.'


 ● Error: The build command produced an exit code greater than 0 - Check the logs
```
Thank you.

**Comment by userdocs** [comment]
Why are you using boost 1.77.0?

**Comment by NickHo99** [comment]
There are some problems when I build with boost 1.78.0 + my customized libtorrent patch
 `./qbittorrent-nox-static.sh -qt release-4.4.0 -lt v2.0.5 all` with all default also not working.

**Comment by NickHo99** [comment]
repo libtorrent patch + boost 1.78.0, libtorrent v2.0.5 also doesn't work, I think it is not related to the issues.

libtorrent 2.0.5-release does not have file 'mmap_disk_job.cpp' but it can be found in it in RC_2_0

command:
```
git clone --depth 1 --branch release-4.4.0_v2.0.5 https://github.com/userdocs/qbittorrent-nox-static
cd qbittorrent-nox-static/
./qbittorrent-nox-static.sh -qt release-4.4.0 -lt v2.0.5 all
```
error:
```
 + Installing libtorrent

Cloning into '/root/qbittorrent-nox-static/qbt-build/libtorrent'...
remote: Enumerating objects: 1005, done.
remote: Counting objects: 100% (1005/1005), done.
remote: Compressing objects: 100% (925/925), done.
remote: Total 1005 (delta 297), reused 232 (delta 72), pack-reused 0
Receiving objects: 100% (1005/1005), 4.15 MiB | 16.21 MiB/s, done.
Resolving deltas: 100% (297/297), done.
Submodule 'deps/asio-gnutls' (https://github.com/paullouisageneau/boost-asio-gnutls.git) registered for path 'deps/asio-gnutls'
Submodule 'deps/try_signal' (https://github.com/arvidn/try_signal.git) registered for path 'deps/try_signal'
Submodule 'simulation/libsimulator' (https://github.com/arvidn/libsimulator.git) registered for path 'simulation/libsimulator'
Cloning into '/root/qbittorrent-nox-static/qbt-build/libtorrent/deps/asio-gnutls'...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (32/32), done.
remote: Compressing objects: 100% (27/27), done.
remote: Total 32 (delta 7), reused 12 (delta 3), pack-reused 0
Receiving objects: 100% (32/32), 19.41 KiB | 1.49 MiB/s, done.
Resolving deltas: 100% (7/7), done.
Cloning into '/root/qbittorrent-nox-static/qbt-build/libtorrent/deps/try_signal'...
remote: Enumerating objects: 19, done.
remote: Counting objects: 100% (19/19), done.
remote: Compressing objects: 100% (18/18), done.
remote: Total 19 (delta 4), reused 2 (delta 0), pack-reused 0
Receiving objects: 100% (19/19), 12.19 KiB | 12.19 MiB/s, done.
Resolving deltas: 100% (4/4), done.
Cloning into '/root/qbittorrent-nox-static/qbt-build/libtorrent/simulation/libsimulator'...
remote: Enumerating objects: 61, done.
remote: Counting objects: 100% (61/61), done.
remote: Compressing objects: 100% (58/58), done.
remote: Total 61 (delta 16), reused 9 (delta 0), pack-reused 0
Receiving objects: 100% (61/61), 165.03 KiB | 343.00 KiB/s, done.
Resolving deltas: 100% (16/16), done.
Submodule path 'deps/asio-gnutls': checked out 'a57d4d36923c5fafa9698e14be16b8bc2913700a'
remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
remote: Enumerating objects: 5, done.
remote: Counting objects: 100% (5/5), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 3 (delta 2), reused 1 (delta 0), pack-reused 0
Unpacking objects: 100% (3/3), 357 bytes | 357.00 KiB/s, done.
From https://github.com/arvidn/try_signal
 * branch            334fd139e2bb387017b42d36753a03935e3bca75 -> FETCH_HEAD
Submodule path 'deps/try_signal': checked out '334fd139e2bb387017b42d36753a03935e3bca75'
Submodule path 'simulation/libsimulator': checked out '64fb5b4fde1879abc09c018604d57e485a12e999'

 ✔ Using 2.0.5 existing patch file

 ✔ Using libtorrent branch master Jamfile file

patching file include/libtorrent/aux_/allocating_handler.hpp
patch unexpectedly ends in middle of line
Hunk #1 succeeded at 124 with fuzz 1.
CXXFLAGS = -std=c++17 -static -w  -I/root/qbittorrent-nox-static/qbt-build/include
LDFLAGS = -static -L/root/qbittorrent-nox-static/qbt-build/lib
OS = LINUX
building boost from source directory:  /root/qbittorrent-nox-static/qbt-build/boost
Performing configuration checks

    - default address-model    : 64-bit [1]
    - default architecture     : x86 [1]
error: Unable to find file or target named
error:     'src/mmap_disk_job.cpp'
error: referred to from project at
error:     '.'


 ● Error: The build command produced an exit code greater than 0 - Check the logs

```
libtorrent file list:
```
ls -al ~/qbittorrent-nox-static/qbt-build/libtorrent/src/
total 2940
drwxr-xr-x  4 root root   4096 Feb  7 18:30 .
drwxr-xr-x 16 root root   4096 Feb  7 18:30 ..
-rw-r--r--  1 root root   3883 Feb  7 18:30 add_torrent_params.cpp
-rw-r--r--  1 root root  85322 Feb  7 18:30 alert.cpp
-rw-r--r--  1 root root   4407 Feb  7 18:30 alert_manager.cpp
-rw-r--r--  1 root root   8510 Feb  7 18:30 announce_entry.cpp
-rw-r--r--  1 root root  10414 Feb  7 18:30 assert.cpp
-rw-r--r--  1 root root   3498 Feb  7 18:30 bandwidth_limit.cpp
-rw-r--r--  1 root root   5982 Feb  7 18:30 bandwidth_manager.cpp
-rw-r--r--  1 root root   2797 Feb  7 18:30 bandwidth_queue_entry.cpp
-rw-r--r--  1 root root  32199 Feb  7 18:30 bdecode.cpp
-rw-r--r--  1 root root   7025 Feb  7 18:30 bitfield.cpp
-rw-r--r--  1 root root   2987 Feb  7 18:30 bloom_filter.cpp
-rw-r--r--  1 root root 107882 Feb  7 18:30 bt_peer_connection.cpp
-rw-r--r--  1 root root   5216 Feb  7 18:30 chained_buffer.cpp
-rw-r--r--  1 root root  11848 Feb  7 18:30 choker.cpp
-rw-r--r--  1 root root   8276 Feb  7 18:30 close_reason.cpp
-rw-r--r--  1 root root   4552 Feb  7 18:30 cpuid.cpp
-rw-r--r--  1 root root   4586 Feb  7 18:30 crc32c.cpp
-rw-r--r--  1 root root  29079 Feb  7 18:30 create_torrent.cpp
-rw-r--r--  1 root root   3357 Feb  7 18:30 directory.cpp
-rw-r--r--  1 root root   7029 Feb  7 18:30 disabled_disk_io.cpp
-rw-r--r--  1 root root   2350 Feb  7 18:30 disk_buffer_holder.cpp
-rw-r--r--  1 root root   6627 Feb  7 18:30 disk_buffer_pool.cpp
-rw-r--r--  1 root root   1849 Feb  7 18:30 disk_interface.cpp
-rw-r--r--  1 root root   3875 Feb  7 18:30 disk_io_job.cpp
-rw-r--r--  1 root root   6706 Feb  7 18:30 disk_io_thread_pool.cpp
-rw-r--r--  1 root root   6726 Feb  7 18:30 disk_job_fence.cpp
-rw-r--r--  1 root root   3416 Feb  7 18:30 disk_job_pool.cpp
drwxr-xr-x  2 root root   4096 Feb  7 18:30 ed25519
-rw-r--r--  1 root root  17775 Feb  7 18:30 entry.cpp
-rw-r--r--  1 root root  45138 Feb  7 18:30 enum_net.cpp
-rw-r--r--  1 root root  11003 Feb  7 18:30 error_code.cpp
-rw-r--r--  1 root root  15008 Feb  7 18:30 escape_string.cpp
-rw-r--r--  1 root root   5080 Feb  7 18:30 ffs.cpp
-rw-r--r--  1 root root   8546 Feb  7 18:30 file.cpp
-rw-r--r--  1 root root   6397 Feb  7 18:30 file_progress.cpp
-rw-r--r--  1 root root  49266 Feb  7 18:30 file_storage.cpp
-rw-r--r--  1 root root   9667 Feb  7 18:30 file_view_pool.cpp
-rw-r--r--  1 root root   3224 Feb  7 18:30 fingerprint.cpp
-rw-r--r--  1 root root   2219 Feb  7 18:30 generate_peer_id.cpp
-rw-r--r--  1 root root   7705 Feb  7 18:30 gzip.cpp
-rw-r--r--  1 root root   7728 Feb  7 18:30 hasher.cpp
-rw-r--r--  1 root root  14440 Feb  7 18:30 hash_picker.cpp
-rw-r--r--  1 root root   3010 Feb  7 18:30 hex.cpp
-rw-r--r--  1 root root  23800 Feb  7 18:30 http_connection.cpp
-rw-r--r--  1 root root  20099 Feb  7 18:30 http_parser.cpp
-rw-r--r--  1 root root  16829 Feb  7 18:30 http_seed_connection.cpp
-rw-r--r--  1 root root  19215 Feb  7 18:30 http_tracker_connection.cpp
-rw-r--r--  1 root root   3903 Feb  7 18:30 i2p_stream.cpp
-rw-r--r--  1 root root  11493 Feb  7 18:30 identify_client.cpp
-rw-r--r--  1 root root   4840 Feb  7 18:30 instantiate_connection.cpp
-rw-r--r--  1 root root   8904 Feb  7 18:30 ip_filter.cpp
-rw-r--r--  1 root root   3905 Feb  7 18:30 ip_helpers.cpp
-rw-r--r--  1 root root  12987 Feb  7 18:30 ip_notifier.cpp
-rw-r--r--  1 root root   6564 Feb  7 18:30 ip_voter.cpp
drwxr-xr-x  2 root root   4096 Feb  7 18:30 kademlia
-rw-r--r--  1 root root   2450 Feb  7 18:30 listen_socket_handle.cpp
-rw-r--r--  1 root root   9186 Feb  7 18:30 lsd.cpp
-rw-r--r--  1 root root  10597 Feb  7 18:30 magnet_uri.cpp
-rw-r--r--  1 root root  14127 Feb  7 18:30 merkle.cpp
-rw-r--r--  1 root root  34963 Feb  7 18:30 merkle_tree.cpp
-rw-r--r--  1 root root  18725 Feb  7 18:30 mmap.cpp
-rw-r--r--  1 root root  58898 Feb  7 18:30 mmap_disk_io.cpp
-rw-r--r--  1 root root  26770 Feb  7 18:30 mmap_storage.cpp
-rw-r--r--  1 root root  24093 Feb  7 18:30 natpmp.cpp
-rw-r--r--  1 root root   5499 Feb  7 18:30 packet_buffer.cpp
-rw-r--r--  1 root root   5967 Feb  7 18:30 parse_url.cpp
-rw-r--r--  1 root root  13130 Feb  7 18:30 part_file.cpp
-rw-r--r--  1 root root  29209 Feb  7 18:30 path.cpp
-rw-r--r--  1 root root  11510 Feb  7 18:30 pe_crypto.cpp
-rw-r--r--  1 root root   4289 Feb  7 18:30 peer_class.cpp
-rw-r--r--  1 root root   2565 Feb  7 18:30 peer_class_set.cpp
-rw-r--r--  1 root root 202884 Feb  7 18:30 peer_connection.cpp
-rw-r--r--  1 root root   9540 Feb  7 18:30 peer_connection_handle.cpp
-rw-r--r--  1 root root   3822 Feb  7 18:30 peer_info.cpp
-rw-r--r--  1 root root  39156 Feb  7 18:30 peer_list.cpp
-rw-r--r--  1 root root   5487 Feb  7 18:30 performance_counters.cpp
-rw-r--r--  1 root root 122078 Feb  7 18:30 piece_picker.cpp
-rw-r--r--  1 root root   2791 Feb  7 18:30 platform_util.cpp
-rw-r--r--  1 root root  14114 Feb  7 18:30 posix_disk_io.cpp
-rw-r--r--  1 root root  13937 Feb  7 18:30 posix_part_file.cpp
-rw-r--r--  1 root root  17713 Feb  7 18:30 posix_storage.cpp
-rw-r--r--  1 root root   1876 Feb  7 18:30 proxy_base.cpp
-rw-r--r--  1 root root   2635 Feb  7 18:30 proxy_settings.cpp
-rw-r--r--  1 root root  38087 Feb  7 18:30 puff.cpp
-rw-r--r--  1 root root   6127 Feb  7 18:30 random.cpp
-rw-r--r--  1 root root  15701 Feb  7 18:30 read_resume_data.cpp
-rw-r--r--  1 root root  10260 Feb  7 18:30 receive_buffer.cpp
-rw-r--r--  1 root root  11172 Feb  7 18:30 request_blocks.cpp
-rw-r--r--  1 root root   4597 Feb  7 18:30 resolve_links.cpp
-rw-r--r--  1 root root   5228 Feb  7 18:30 resolver.cpp
-rw-r--r--  1 root root   2704 Feb  7 18:30 session_call.cpp
-rw-r--r--  1 root root  18000 Feb  7 18:30 session.cpp
-rw-r--r--  1 root root  36445 Feb  7 18:30 session_handle.cpp
-rw-r--r--  1 root root 220910 Feb  7 18:30 session_impl.cpp
-rw-r--r--  1 root root   8070 Feb  7 18:30 session_params.cpp
-rw-r--r--  1 root root   2272 Feb  7 18:30 session_settings.cpp
-rw-r--r--  1 root root  21956 Feb  7 18:30 session_stats.cpp
-rw-r--r--  1 root root  31855 Feb  7 18:30 settings_pack.cpp
-rw-r--r--  1 root root   9721 Feb  7 18:30 sha1.cpp
-rw-r--r--  1 root root   6118 Feb  7 18:30 sha1_hash.cpp
-rw-r--r--  1 root root   4886 Feb  7 18:30 sha256.cpp
-rw-r--r--  1 root root  10063 Feb  7 18:30 smart_ban.cpp
-rw-r--r--  1 root root   4634 Feb  7 18:30 socket_io.cpp
-rw-r--r--  1 root root   7521 Feb  7 18:30 socket_type.cpp
-rw-r--r--  1 root root   2824 Feb  7 18:30 socks5_stream.cpp
-rw-r--r--  1 root root   6136 Feb  7 18:30 ssl.cpp
-rw-r--r--  1 root root   4137 Feb  7 18:30 stack_allocator.cpp
-rw-r--r--  1 root root   4676 Feb  7 18:30 stat_cache.cpp
-rw-r--r--  1 root root   1890 Feb  7 18:30 stat.cpp
-rw-r--r--  1 root root  20155 Feb  7 18:30 storage_utils.cpp
-rw-r--r--  1 root root  10856 Feb  7 18:30 string_util.cpp
-rw-r--r--  1 root root   1756 Feb  7 18:30 time.cpp
-rw-r--r--  1 root root   3350 Feb  7 18:30 timestamp_history.cpp
-rw-r--r--  1 root root 349912 Feb  7 18:30 torrent.cpp
-rw-r--r--  1 root root  26966 Feb  7 18:30 torrent_handle.cpp
-rw-r--r--  1 root root  54042 Feb  7 18:30 torrent_info.cpp
-rw-r--r--  1 root root   4241 Feb  7 18:30 torrent_peer_allocator.cpp
-rw-r--r--  1 root root   8307 Feb  7 18:30 torrent_peer.cpp
-rw-r--r--  1 root root   2552 Feb  7 18:30 torrent_status.cpp
-rw-r--r--  1 root root  14951 Feb  7 18:30 tracker_manager.cpp
-rw-r--r--  1 root root  27931 Feb  7 18:30 udp_socket.cpp
-rw-r--r--  1 root root  22172 Feb  7 18:30 udp_tracker_connection.cpp
-rw-r--r--  1 root root  43747 Feb  7 18:30 upnp.cpp
-rw-r--r--  1 root root   5437 Feb  7 18:30 utf8.cpp
-rw-r--r--  1 root root  19808 Feb  7 18:30 ut_metadata.cpp
-rw-r--r--  1 root root  20340 Feb  7 18:30 ut_pex.cpp
-rw-r--r--  1 root root   9721 Feb  7 18:30 utp_socket_manager.cpp
-rw-r--r--  1 root root 100224 Feb  7 18:30 utp_stream.cpp
-rw-r--r--  1 root root   1634 Feb  7 18:30 version.cpp
-rw-r--r--  1 root root   6285 Feb  7 18:30 web_connection_base.cpp
-rw-r--r--  1 root root  40164 Feb  7 18:30 web_peer_connection.cpp
-rw-r--r--  1 root root  12607 Feb  7 18:30 write_resume_data.cpp
-rw-r--r--  1 root root   5022 Feb  7 18:30 xml_parse.cpp
```
[log.zip](https://github.com/userdocs/qbittorrent-nox-static/files/8017852/log.zip)

**Comment by userdocs** [comment]
I think if you build using cmake it will work.

`-c`

**Comment by userdocs** [comment]
btw I know what the problem is so thanks for bringing to my attention. It missed my testing since I was using cmake.

I've resolved it using a jamfle patch for non cmake 2.0.5 builds here.

https://github.com/userdocs/qbittorrent-nox-static/blob/master/qbt-build/patches/libtorrent/2.0.5/Jamfile

The problem will go away with libtorrent v2.0.6 when released.

But if cmake builds are fine then you can avoid the problem that way
